tools/screenshots.sh

```diff
@@ -0,0 +1,748 @@
+  # In this version this is simulated and strings are embedded in the code
+  TEXT_en_US_MYSITE="Manage your site\neverywhere you go"
+  TEXT_en_US_READER="Enjoy your\nfavourite sites"
+  TEXT_en_US_NOTIFS="Get notified\nin real-time"
+
+  TEXT_el_GR_MYSITE="Έλεγξε τον ιστότοπο σου\nαπ'οπουδήποτε βρίσκεσαι"
+  TEXT_el_GR_READER="Απόλαυσε τις\nαγαπημένες σου σελίδες"
+  TEXT_el_GR_NOTIFS="Ειδοποιήσεις σε\nπραγματικό χρόνο"
+
+  TEXT_it_IT_MYSITE="Gestisci il tuo sito\novunque tu sia"
+  TEXT_it_IT_READER="Leggi i tuoi\nsiti preferiti"
+  TEXT_it_IT_NOTIFS="Rimani aggiornato\nin tempo reale"
+}
+
+# Translates language codes
+function glotPress_googlePlay() {
+  lang=$1
+  index=0
+  for data in "${LangGlotPress[@]}"; do
+    if [ $data == $lang ]; then
+      return $index
+    fi
+    index=$((index+1))
+  done
+
+  return 255
+}
+
+# Processes the screenshot for the provided device
+function process_device() {
+  device=$1
+
+  for line in $(cat $LANG_FILE); do
+    cod=$(echo $line|cut -d "," -f1|tr -d " ")
+    glotPress_googlePlay $cod
+    locIdx=$?
+
+    if [ $locIdx -lt 255 ]; then
+      locLang=${LangGooglePlay[$locIdx]}
+      require_image_dirs $PROD_DIR $device $locLang
+
+      for screen in ${SCREENS[*]}; do
+        process_screenshot $device $locLang $screen
+      done
+
+    else
+      show_warning_message "Skipping language $cod as no Google Play pair was found."
+      warningCount=$((warningCount+1))
+    fi
+  done
+}
+
+# Takes the screenshot for the provided device
+function screenshot_device() {
+  device=$1
+
+  kill_emus
+  start_emu $device
+  wait_emu
+  uninstall
+  install
+  login
+
+  kill_app # kill the app so when restarting we don't have any first-time launch effects like promo dialogs and such
+
+  start_app
+
+  kill_app # kill the app so when restarting we don't have any first-time launch effects like promo dialogs and such
+
+  for line in $(cat $LANG_FILE); do
+    cod=$(echo $line|cut -d "," -f1|tr -d " ")
+    glotPress_googlePlay $cod
+    locIdx=$?
+
+    if [ $locIdx -lt 255 ]; then
+      locLang=${LangGooglePlay[$locIdx]}
+      require_image_dirs $TAKE_DIR $device $locLang 
+
+      locale $locLang
+
+      start_app
+
+      for screen in ${SCREENS[*]}; do
+        coords=$device\_COORDS_$screen
+        tap_on ${!coords}
+
+        geekytime
+        take_screenshot $device $locLang $screen
+      done
+
+    else
+      show_warning_message "Skipping language $cod as no Google Play pair was found."
+      warningCount=$((warningCount+1))
+    fi
+  done
+}
+
+# Takes the required screenshots
+function execute_take() {
+  check_params
+  show_title_message "Starting the screenshot taker process..."
+  warningCount=0
+  if [ $RUN_DEV == "ALL" ]; then
+    for device in ${DEVICES[*]}; do
+      screenshot_device $device
+    done
+  else 
+    screenshot_device $RUN_DEV
+  fi
+
+  if [ $CMD == $CMD_TAKE ]; then
+    if [ $warningCount -eq 0]; then
```

Comment:
There is a missing space here before the `]` that breaks the test.

Reply:
Ouch! I remember having fixed it and even having tested the fix. How I managed to loose the change, I don't know :-( 
Maybe I'm getting old... :-)